src/binder_call_forwarding.c

```diff
@@ -273,29 +281,58 @@ binder_call_forwarding_query_ok(
     GBinderReader reader;
     gsize count = 0;
 
-    /* getCallForwardStatusResponse(RadioResponseInfo, vec<CallForwardInfo>) */
     gbinder_reader_copy(&reader, args);
-    infos = gbinder_reader_read_hidl_type_vec(&reader, RadioCallForwardInfo,
-        &count);
-    if (count) {
-        gsize i;
-
-        list = g_new0(struct ofono_call_forwarding_condition, count);
-        for (i = 0; i < count; i++) {
-            const RadioCallForwardInfo* info = infos + i;
-            struct ofono_call_forwarding_condition* fw = list + i;
-
-            fw->status = info->status;
-            fw->cls = info->serviceClass;
-            fw->time = info->timeSeconds;
-            fw->phone_number.type = info->toa;
-            memcpy(fw->phone_number.number, info->number.data.str,
-                MIN(OFONO_MAX_PHONE_NUMBER_LENGTH, info->number.len));
+
+    if (cbd->self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+        /* getCallForwardStatusResponse(RadioResponseInfo, vec<CallForwardInfo>) */
+        const RadioCallForwardInfo* infos = gbinder_reader_read_hidl_type_vec(
+            &reader, RadioCallForwardInfo, &count);
+        if (count) {
+            gsize i;
+
+            list = g_new0(struct ofono_call_forwarding_condition, count);
+            for (i = 0; i < count; i++) {
+                const RadioCallForwardInfo* info = infos + i;
+                struct ofono_call_forwarding_condition* fw = list + i;
+
+                fw->status = info->status;
+                fw->cls = info->serviceClass;
+                fw->time = info->timeSeconds;
+                fw->phone_number.type = info->toa;
+                memcpy(fw->phone_number.number, info->number.data.str,
+                    MIN(OFONO_MAX_PHONE_NUMBER_LENGTH, info->number.len));
+            }
+        }
+    } else {
+        /* getCallForwardStatusResponse(RadioResponseInfo, CallForwardInfo[] callForwardInfos) */
+        gbinder_reader_read_uint32(&reader, (guint32 *)&count);
+        if (count) {
+            gsize i;
+
+            list = g_new0(struct ofono_call_forwarding_condition, count);
+            for (i = 0; i < count; i++) {
+                struct ofono_call_forwarding_condition* fw = list + i;
+
+                // Non-null parcelable
+                gbinder_reader_read_int32(&reader, NULL);
+                // Parcelable size
+                gbinder_reader_read_int32(&reader, NULL);
+                gbinder_reader_read_int32(&reader, &fw->status);
+                gbinder_reader_read_int32(&reader, NULL);
+                gbinder_reader_read_int32(&reader, &fw->cls);
+                gbinder_reader_read_int32(&reader, &fw->phone_number.type);
+                gchar* number = gbinder_reader_read_string16(&reader);
+                if (number) {
+                    memcpy(fw->phone_number.number, number,
+                        MIN(OFONO_MAX_PHONE_NUMBER_LENGTH, strlen(number)));
+                    g_free(number);
```

Comment:
`g_free()` does handle `NULL` so it is not necessary for this to be within the `if()`.

Reply:
I'll leave it like it is since the if will be there anyway so no extra if is being added for g_free.